pkg/cleanup/cni.go

```diff
@@ -4,47 +4,28 @@ import (
 	"fmt"
 	"os"
 
-	"github.com/k0sproject/k0s/internal/pkg/file"
 	"github.com/sirupsen/logrus"
 )
 
-type cni struct {
-	Config   *Config
-	toRemove []string
-}
+type cni struct{}
 
 // Name returns the name of the step
 func (c *cni) Name() string {
 	return "CNI leftovers cleanup step"
 }
 
-// NeedsToRun checks if there are and CNI leftovers
-func (c *cni) NeedsToRun() bool {
+// Run removes found CNI leftovers
+func (c *cni) Run() error {
+	var msg []error
+
 	files := []string{
 		"/etc/cni/net.d/10-calico.conflist",
 		"/etc/cni/net.d/calico-kubeconfig",
 		"/etc/cni/net.d/10-kuberouter.conflist",
 	}
-
-	for _, f := range files {
-		if file.Exists(f) {
-			c.toRemove = append(c.toRemove, f)
-		}
-	}
-	return len(c.toRemove) > 0
-}
-
-// Run removes found CNI leftovers
-func (c *cni) Run() error {
-	return removeCNILeftovers(c.toRemove)
-}
-
-func removeCNILeftovers(files []string) error {
-	var msg []error
-
 	for _, f := range files {
-		if file.Exists(f) {
-			if err := os.Remove(f); err != nil {
+		if err := os.Remove(f); err != nil {
+			if !os.IsNotExist(err) {
 				logrus.Debug("failed to remove", f, err)
 				msg = append(msg, err)
 			}
```

Comment:
```suggestion
		if err := os.Remove(f); err != nil && !os.IsNotExist(err) {
			logrus.Debug("failed to remove", f, err)
			msg = append(msg, err)
```

Reply:
Seems like we should use `errors.Is(fs.ErrNotExist)`

From https://pkg.go.dev/os#IsNotExist
> This function predates errors.Is. It only supports errors returned by the os package. New code should use errors.Is(err, fs.ErrNotExist).